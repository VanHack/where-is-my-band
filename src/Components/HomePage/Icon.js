import styled from "styled-components";

const Icon = styled.span`
  color: ${props => (props.primary ? props.theme.primary : "white")};
  font-size: ${props => (props.big ? "80px" : "20px")};
  padding-right: ${props => (props.left ? "15px" : "0px")};
  padding-left: ${props => (props.right ? "15px" : "0px")};
`;

export default Icon;
