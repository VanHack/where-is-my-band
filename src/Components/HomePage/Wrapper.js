import styled from "styled-components";

const Wrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: flex-start;
  justify-content: center;
`;

export default Wrapper;
