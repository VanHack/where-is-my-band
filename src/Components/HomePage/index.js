import React from "react";

import NotFound from "./NotFound";
import Wrapper from "./Wrapper";
import Icon from "./Icon";
import Start from "../Start";
import SearchBar from "../SearchBar";
import Artist from "../Artist";
import EventList from "../EventList";

const Result = ({ artist, events, setActiveEvent }) => {
  // checking if the search yield any result
  if (artist.found === false) {
    return (
      <NotFound>
        Sorry, Artist/Band not found
        <Icon big primary className="fa fa-frown-o" left />
      </NotFound>
    );
  }

  // this will be false if no search has been done
  if (!artist.name) {
    return <Start />;
  }

  // rendering results
  return (
    <Wrapper>
      <Artist artist={artist} />
      <EventList events={events} setActiveEvent={setActiveEvent} />
    </Wrapper>
  );
};

const HomePage = ({ search, artist, events, setActiveEvent }) => {
  return (
    <div>
      <SearchBar search={search} />
      <Result artist={artist} events={events} setActiveEvent={setActiveEvent} />
    </div>
  );
};

export default HomePage;
