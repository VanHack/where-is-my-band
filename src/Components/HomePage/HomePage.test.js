import React from 'react';
import HomePage from './index';
import renderer from 'react-test-renderer';

test('renders the component according to snapshot', () => {
  const myEvents = [{
    id: '13',
    venue: {
      name: 'MyVenue',
      datetime: new Date(),
      on_sale_datetime: new Date()
    }
  }]

  const rockStar = {
    image_url: 'http://example.com/some-image.jpg',
    name: 'Rock Star!',
    excerpt: 'Rock Star band description',
    facebook_page_url: 'http://facebook.com/rockstar',
  }

  const component = renderer.create(
    <HomePage
      search={'Rock Star!'}
      artist={rockStar}
      events={myEvents}
      setActiveEvent={myEvents[0]} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
