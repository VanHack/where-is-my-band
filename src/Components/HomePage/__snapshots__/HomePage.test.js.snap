// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component according to snapshot 1`] = `
<div>
  <form
    className="sc-bwzfXH ioSiIW"
    onSubmit={[Function]}
  >
    <input
      className="sc-htpNat PsiW"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
    <button
      aria-hidden="true"
      className="fa fa-search sc-bxivhb bPqXOW"
      type="submit"
    />
  </form>
  <div
    className="sc-bdVaJa dIvFmw"
  >
    <div
      className="sc-ifAKCX fUvxRh"
    >
      <img
        className="sc-bZQynM kkhJKC"
        src="http://example.com/some-image.jpg"
      />
      <h1
        className="sc-gzVnrw jwBRWH"
      >
        Rock Star!
      </h1>
      <div
        className="sc-EHOje eMEULM"
      >
        Rock Star band description
      </div>
      <a
        className="sc-htoDjs fgpYNQ"
        href="http://facebook.com/rockstar"
        target="_blank"
      >
        Follow on Facebook
      </a>
    </div>
    <div
      className="sc-dnqmqq kVgMYi"
    >
      <div
        className="sc-jTzLTM jaSgVr"
      >
        <h1
          className="sc-fjdhpX icxhmf"
        >
          MyVenue
        </h1>
        <div
          className="sc-jzJRlG eqqNZe"
        >
          undefined, undefined
        </div>
        <div
          className="sc-cSHVUG dHFzSm"
        >
          Date: 
          Invalid Date
        </div>
        <div
          className="sc-kAzzGY fLdUSY"
        >
          Tickets available:
           TBA
        </div>
      </div>
      <div
        className="sc-iwsKbI lauawB"
      />
    </div>
  </div>
</div>
`;
